vpr/src/pack/attraction_groups.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * attraction_groups.h
+ *
+ *  Created on: Jun. 13, 2021
+ *      Author: khalid88
+ */
+
+#ifndef VPR_SRC_PACK_ATTRACTION_GROUPS_H_
+#define VPR_SRC_PACK_ATTRACTION_GROUPS_H_
+
+#include "vtr_strong_id.h"
+#include "vtr_vector.h"
+#include "atom_netlist.h"
+#include "globals.h"
+
+/**
+ * @file
+ * @brief This file defines the AttractionInfo class, which is used to store atoms in attraction groups, which are
+ *        used during the clustering process.
+ *
+ * Overview
+ * ========
+ * Attraction groups are used during the clustering process to tell the clusterer which atoms should be packed
+ * in the same cluster by virtue of having the same floorplan constraints. The attraction groups may also be
```

Comment:
rephrase more precisely - "atoms in the same att group will be highly desirable to be packed together. Will be seen as having higher gain if in the same att group. Right now the att groups are based on floorplan constraints, atoms in the same partition are in the same att group. Right now used for floorplanning, but could be used for on other concepts in the future